src/metabase/models/card.clj

```diff
@@ -321,11 +366,11 @@
                                        :entity_id    true})
           ;; Make sure we normalize the query before calling `pre-update` or `pre-insert` because some of the
           ;; functions those fns call assume normalized queries
-          :pre-update     (comp populate-query-fields pre-update populate-result-metadata maybe-normalize-query)
-          :pre-insert     (comp populate-query-fields pre-insert populate-result-metadata maybe-normalize-query)
+          :pre-update     (comp migrate-viz-settings populate-query-fields pre-update populate-result-metadata maybe-normalize-query)
+          :pre-insert     (comp migrate-viz-settings populate-query-fields pre-insert populate-result-metadata maybe-normalize-query)
           :post-insert    post-insert
           :pre-delete     pre-delete
-          :post-select    public-settings/remove-public-uuid-if-public-sharing-is-disabled}))
+          :post-select    post-select}))
```

Comment:
Small nit:

The pre-update/insert keys are using `comp`, but the other keys are using private fns with thread-first macros. Is there a reason for the difference?